apps/src/templates/VersionHistory.jsx

```diff
@@ -0,0 +1,92 @@
+var React = require('react');
+var VersionRow = require('./VersionRow.jsx');
+var sourcesApi = require('../clientApi')('sources');
+
+/**
+ * A component for viewing project version history.
+ */
+module.exports = React.createClass({
+
+  getInitialState: function () {
+    return {
+      versions: null,
+      statusMessage: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    sourcesApi.ajax('GET', 'main.json/versions', this.onVersionListReceived, this.onAjaxFailure);
```

Comment:
I'm not necessarily worried about dots in the path in general. It's more that it looks when I see what appears to be a file (main.json), but that's not the end of the path. Maybe this isn't as strange as it feels to me.